File: Backend/Core.Domain/Document.cs
using Framework.Domain;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Core.Domain
{
    public class Document : EntityBase
    {

        [Required(ErrorMessage = "O tipo do documento é obrigatório")]
        [ForeignKey(nameof(FKDocumentType))]
        public int IdDocumentType { get; set; }
        public DocumentType FKDocumentType { get; set; }

        [Required(ErrorMessage = "O documento é obrigatório")]
        [MaxLength(30)]
        public string Value { get; set; }

        [Required(ErrorMessage = "O Id da pessoa é obrigatório")]
        [ForeignKey(nameof(FKPerson))]
        public int IdPerson { get; set; }
        public Person FKPerson { get; set; }

    }
}

File: Frontend/src/app/system/_domains/Document.ts
import { Injectable } from '@angular/core';
import { FormBuilder, FormGroup, Validators, FormArray } from '@angular/forms';

import { _EntityBase } from './_EntityBase';
import { Person } from './Person';
import { DocumentType } from './DocumentType';

@Injectable()
export class Document extends _EntityBase {
  public IdDocumentType: number;
  public FKDocumentType: DocumentType;
  public Value: string;
  public IdPerson: number;
  public FKPerson: Person;

  constructor() {
    super();
    this.FKPerson = new Person();
  }
}
